From: SeongJae Park <sj@kernel.org>
Date: Sun, 22 Dec 2024 12:13:12 -0800
Subject: [PATCH] ==== inclusive damos_filter ====

mm/damon: extend DAMOS filters for allowing specific types of memory

DAMOS fitlers are exclusive filters.  It only denies applying a given
DAMOS action to specific types of memory.  This has below problems.

First, the name itself is not explicit enough.  This actually resulted
in confusions[1] before, and we tried to improve the situation by
improving the documentation and adding clear options to DAMON user-space
tool[2].

Secondly, the functionality is restrictive.  Users cannot apply an
action to multiple types of memory.  For example, building a DAMOS
scheme that applies the action to memory that belongs to cgroup A "or"
cgroup B is impossible.  For that, users should add two schemes, each
filtering out memory that not belong to cgroup A and cgroup B,
respectively.  This is cumbersome but may work for DAMOS schemes that
intended to make system behavioral changes, e.g., DAMOS_PAGEOUT.  But if
it is for page level peroperties based monitoring (want to know how much
of memory of specific access temperature belongs to cgroup A or cgroup
B), getting the number from the two schemes is difficult.

Extend DAMOS filter to be used for not only excluding, but also
including the DAMOS action target memory.

Note that the default behavior is exclusive, so this change will not
disturb old users with unexpected behavioral changes.

[1] https://lore.kernel.org/20240320165619.71478-1-sj@kernel.org
[2] https://git.kernel.org/sj/damo/c/b6a722c85ff91e5abe9dd47135e300df243da056

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 damon_meta_changes/RRsi2zLl | 0
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 damon_meta_changes/RRsi2zLl

diff --git a/damon_meta_changes/RRsi2zLl b/damon_meta_changes/RRsi2zLl
new file mode 100644
index 000000000000..e69de29bb2d1
-- 
2.39.5

